packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -123,6 +126,79 @@ void _debugRecordDownStream(String channelTypeName, String name,
   _debugLaunchProfilePlatformChannels();
 }
 
+class BackgroundIsolateBinaryMessenger extends BinaryMessenger {
+  final ReceivePort receivePort = ReceivePort();
+  final Map<int, Completer<ByteData?>> _completers = <int, Completer<ByteData?>>{};
+  int _messageCount = 0;
+
+  static BinaryMessenger? _backgroundIsolateBinaryMessenger;
+
+  static void ensureInitialized(ui.RootIsolateToken token) {
+    if (_backgroundIsolateBinaryMessenger == null) {
+      ui.PlatformDispatcher.instance.registerBackgroundIsolate(token);
+      final BackgroundIsolateBinaryMessenger portBinaryMessenger = BackgroundIsolateBinaryMessenger();
+      _backgroundIsolateBinaryMessenger = portBinaryMessenger;
+      portBinaryMessenger.receivePort.listen((dynamic message) {
+        try {
+          final List<dynamic> args = message as List<dynamic>;
+          final int identifier = args[0] as int;
+          final Uint8List bytes = args[1] as Uint8List;
+          final ByteData byteData = ByteData.sublistView(bytes);
+          final Completer<ByteData?> completer =
+              portBinaryMessenger._completers[identifier]!;
+          portBinaryMessenger._completers.remove(identifier);
+          completer.complete(byteData);
```

Comment:
You can simplify this to:
```suggestion
          _completers.remove(identifier)!.complete(byteData);
```

Reply:
done